nifi-extension-bundles/nifi-protobuf-bundle/nifi-protobuf-services/src/main/java/org/apache/nifi/services/protobuf/StandardProtobufReader.java

```diff
@@ -170,6 +174,84 @@ protected PropertyDescriptor buildSchemaTextProperty() {
+            }
+
+            if (schemaTextValue == null || schemaTextValue.isBlank()) {
+                problems.add(new ValidationResult.Builder()
+                    .subject(SCHEMA_TEXT.getDisplayName())
+                    .input(schemaTextValue)
+                    .valid(false)
+                    .explanation("Schema Text cannot be empty when using \"Use 'Schema Text' Property\" strategy")
+                    .build());
+                return problems;
+            }
+
+            try {
+                // Try to compile the schema to validate it's valid protobuf format
+                final String hash = sha256Hex(schemaTextValue);
+                final SchemaIdentifier schemaIdentifier = SchemaIdentifier.builder()
+                    .name(hash + PROTO_EXTENSION)
+                    .build();
+                final Schema compiledSchema = validateSchemaCompiles(schemaIdentifier, schemaTextValue);
+
+                // Validate Message Name if using MESSAGE_NAME_PROPERTY strategy
+                final String messageNameStrategy = validationContext.getProperty(MESSAGE_NAME_RESOLUTION_STRATEGY).getValue();
+                if (MESSAGE_NAME_PROPERTY.getValue().equals(messageNameStrategy)) {
+                    final PropertyValue messageNameProperty = validationContext.getProperty(MESSAGE_NAME);
+                    final String messageNameValue = messageNameProperty.getValue();
+
+                    if (validationContext.isExpressionLanguageSupported(MESSAGE_NAME.getName())
+                        && validationContext.isExpressionLanguagePresent(messageNameValue)) {
+                        return problems;
+                    }
+
+                    if (messageNameValue != null && !messageNameValue.isBlank()) {
+                        // Check if the message name exists in the compiled schema
+                        if (compiledSchema.getType(messageNameValue) == null) {
+                            problems.add(new ValidationResult.Builder()
+                                .subject(MESSAGE_NAME.getDisplayName())
+                                .input(messageNameValue)
+                                .valid(false)
+                                .explanation(String.format("Message name '%s' cannot be found in the provided protobuf schema", messageNameValue))
+                                .build());
+                        }
+                    }
+                }
+
+            } catch (final SchemaCompilationException e) {
+                problems.add(new ValidationResult.Builder()
+                    .subject(SCHEMA_TEXT.getDisplayName())
+                    .input(schemaTextValue)
+                    .valid(false)
+                    .explanation("Invalid protobuf schema format: " + e.getMessage())
+                    .build());
+            }
+        }
+
+        return problems;
+    }
+
+    // Compile the schema to validate it's correct. The method is used only for validation purposes.
```

Comment:
This comment mostly reiterates the method name, and the usage in validation is clear, so recommend removing this comment.